sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/internal/taskdefutil/TaskDefBuilder.java

```diff
@@ -5,15 +5,17 @@
 import io.littlehorse.sdk.common.proto.VariableDef;
 import io.littlehorse.sdk.common.proto.VariableType;
 import java.util.List;
+import java.util.Map;
 
 public class TaskDefBuilder {
 
     public Object executable;
     public String taskDefName;
     public LHTaskSignature signature;
 
-    public TaskDefBuilder(Object executable, String taskDefName) throws TaskSchemaMismatchError {
-        signature = new LHTaskSignature(taskDefName, executable);
+    public TaskDefBuilder(Object executable, String taskDefName, Map<String, String> valuesForPlaceHolders)
```

Comment:
@eduwercamacaro do we expose this class to our clients? And should we? I think it is internal, right?

Reply:
It is internal, and we should not expose this to our clients. However, I suggest reducing the access for this class's constructor and making it final for now.

Reply:
@eduwercamacaro you mean making the class TaskDefBuilder to be final?

Reply:
Yes. However, I think that should be done in a separate Issue / PR because it is not related to Nelson's improvement. We also need to do that for many other classes which we "leak" to our users, so it deserves a GH Issue so that we can address the entire problem rather than just only one class.

@diablouma feel free to merge this!